### ISSUE TITLE
SMTP_FROM possibly ignored 

### ISSUE THREAD

**Issue by rroseselavy42** [cortezaproject/corteza-server#7]
**Describe the bug**
Using the Docker simple instructions a Corteza server instance with SMTP env variables set and carried into the image do not result in working mailing. When signing up I get 

> Error: could not send email authentication notification: gomail: could not send email 1: gomail: invalid address "": mail: no address

**To Reproduce**
download example .env file. 
set SMTP settings
use docker-compose up -d 
sign up for new account
**Expected behavior**
A new account email should be sent. 

**Additional context**
SMTP variables appear to be present in the server environment when I check inside of them. 
Not sure if SMTP_FROM is being ignored or if when signing up the mailto is not being read properly.

**Comment by darh** [comment]
@rroseselavy42 hi, thank you for your interest.

Actually, with SMTP_FROM you set [default from](https://github.com/cortezaproject/corteza-server/blob/master/internal/mail/mail.go#L61) value.

Auth (reg.) emails are generated in [system/internal/service/auth_notification.go](https://github.com/cortezaproject/corteza-server/blob/master/system/internal/service/auth_notification.go#L143) -- looks like your setup did not automatically set the value for email.

Reg. email sender should pick up the value from `auth.mail.from-address` (see [auto-discovery code](https://github.com/cortezaproject/corteza-server/blob/master/system/internal/service/auth_settings.go#L81) for that).

So, 2 problems:
 - why was `auth.mail.from-address` not set on the initial run
 - should add the fallback to `SMTP_FROM` (`internal/mail.defaultFrom`)

----

In the meanwhile, you can set the `auth.mail.from-address` manually:
```sh
docker-compose exec server /bin/corteza-server system settings set auth.mail.from-address '"your-email@example.tld"'
```
(not sure what your setup is, please adapt first part (before settings) to your environment)

**Comment by rroseselavy42** [comment]
I tried to set the aut.mail.from-address and it seems to still be giving me 
```
Error: could not send password reset notification: gomail: could not send email 1: gomail: invalid address "": mail: no address
```

logs from the command I use to set auth.mail.from-address
```
{"level":"info","ts":1565096083.3525136,"logger":"compose.database","caller":"db/connector.go:76","msg":"connected to the database","name":"compose","dsn":"c****a:R****O@tcp(db:3306)/corteza?collation=utf8mb4_general_ci"}
{"level":"info","ts":1565096083.3553052,"logger":"messaging.database","caller":"db/connector.go:76","msg":"connected to the database","name":"messaging","dsn":"c****a:R****O@tcp(db:3306)/corteza?collation=utf8mb4_general_ci"}
{"level":"info","ts":1565096083.3587437,"logger":"system.database","caller":"db/connector.go:76","msg":"connected to the database","name":"system","dsn":"c****a:R****O@tcp(db:3306)/corteza?collation=utf8mb4_general_ci"}
{"level":"info","ts":1565096083.358835,"logger":"compose","caller":"service/service.go:42","msg":"initializing store","path":"/data/compose"}
{"level":"info","ts":1565096083.3596082,"logger":"compose","caller":"service/service.go:45","msg":"initializing store","path":"/data/messaging"}
```

**Comment by rroseselavy42** [comment]
looking at the server logs when I try a password reset.
```
server_1  | 2019/08/06 12:54:30 Returned error cause is not a MySQLError, invalid username/password combination
server_1  | 2019/08/06 12:55:34 Returned error cause is not a MySQLError, invalid username/password combination
server_1  | 2019/08/06 13:01:04 Returned error cause is not a MySQLError, invalid username/password combination
```

**Comment by rroseselavy42** [comment]
hmm, looking at this more I'm wondering if what's not being set is the user email to send TO

**Comment by rroseselavy42** [comment]
NOPE! my bad, just needed to stop and start the server my bad should have tried that first.

**Comment by darh** [comment]
:+1:

**Comment by darh** [comment]
Fixed with 0dceabc5, restart after first run is no longer needed

**Comment by boubafof** [comment]
Hello Team, 
I have the same issue with corteza version version 2021.3.12 , My corteza App is running on ubuntu 20.04 so  I installed corteza and when i first create an account for the admin user, the email is never sent, even though I setup my SMTP on .ENV file  I have the error .
I need a helping Hand
> could not send email: gomail: could not send email 1: 550 Verification failed for <info@example.tld> The mail server does not recognize info@example.tld as a valid sender. Sender verify failed``
